build/ci_utils/src/actions/artifacts.rs

```diff
@@ -96,6 +97,7 @@ pub fn upload_directory(
     let dir = dir.into();
     info!("Uploading directory {}.", dir.display());
     let files = single_dir_provider(&dir);
+    #[allow(clippy::redundant_closure_call)]
```

Comment:
does it work without the closure? Why are we doing it in such a strange way? A docs explaining that should be added here IMO